src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Custom/Native/Call/NativeCallTracerCallFrameConverter.cs

```diff
@@ -0,0 +1,97 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Nethermind.Serialization.Json;
+
+namespace Nethermind.Evm.Tracing.GethStyle.Custom.Native.Call;
+
+public class NativeCallTracerCallFrameConverter : JsonConverter<NativeCallTracerCallFrame>
+{
+    public override void Write(Utf8JsonWriter writer, NativeCallTracerCallFrame value, JsonSerializerOptions options)
+    {
+        NumberConversion? previousValue = ForcedNumberConversion.ForcedConversion.Value;
+        try
+        {
+            writer.WriteStartObject();
+
+            ForcedNumberConversion.ForcedConversion.Value = NumberConversion.Hex;
+            writer.WritePropertyName("type"u8);
+            JsonSerializer.Serialize(writer, value.Type.GetName(), options);
+
+            writer.WritePropertyName("from"u8);
+            JsonSerializer.Serialize(writer, value.From, options);
+
+            if (value.To is not null)
+            {
+                writer.WritePropertyName("to"u8);
+                JsonSerializer.Serialize(writer, value.To, options);
+            }
+
+            if (value.Value is not null)
+            {
+                writer.WritePropertyName("value"u8);
+                JsonSerializer.Serialize(writer, value.Value, options);
+            }
+
+            writer.WritePropertyName("gas"u8);
+            JsonSerializer.Serialize(writer, value.Gas, options);
+
+            writer.WritePropertyName("gasUsed"u8);
+            JsonSerializer.Serialize(writer, value.GasUsed, options);
+
+            writer.WritePropertyName("input"u8);
+            if (value.Input is null)
+            {
+                writer.WriteStringValue("0x"u8);
+            }
+            else
+            {
+                JsonSerializer.Serialize(writer, value.Input.AsSpan().ToArray(), options);
```

Comment:
why those calls are needed? ArrayPoolList should be easily serializable as collection as it implements `IList` interface. And it kind of beats the purpose of using it.

Reply:
The ArrayPoolLists are serializable, however they'll most likely need a new converter to serialize their contents as hex. Currently the `ByteArrayConverter` only supports `byte[]` and `ReadOnlySpan<byte>` types for hex conversions. I'll look into making these changes to avoid needing to convert the ArrayPoolList

Reply:
Paired and resolved synchronously